src/lib/transaction_snark/test/zkapps_examples/initialize_state/initialize_state.ml

```diff
@@ -0,0 +1,271 @@
+        (* TODO: This is a pain. *)
+        let other_parties_hash = Parties.Call_forest.hash ps in
+        let protocol_state_predicate_hash =
+          Zkapp_precondition.Protocol_state.digest protocol_state_precondition
+        in
+        let memo_hash = Signed_command_memo.hash memo in
+        Parties.Transaction_commitment.create ~other_parties_hash
+          ~protocol_state_predicate_hash ~memo_hash
+      in
+      let fee_payer : Party.Fee_payer.t =
+        { body =
+            { Party.Body.Fee_payer.dummy with
+              public_key = pk_compressed
+            ; balance_change = Currency.Fee.(of_int 100)
+            ; protocol_state_precondition
+            }
+        ; authorization = Signature.dummy
+        }
+      in
+      let full_commitment =
+        Parties.Transaction_commitment.with_fee_payer transaction_commitment
+          ~fee_payer_hash:
+            (Parties.Call_forest.Digest.Party.create
+               (Party.of_fee_payer fee_payer))
+      in
+      let sign_all ({ fee_payer; other_parties; memo } : Parties.t) : Parties.t
+          =
+        let fee_payer =
+          match fee_payer with
+          | { body = { public_key; _ }; _ }
+            when Public_key.Compressed.equal public_key pk_compressed ->
+              { fee_payer with
+                authorization =
+                  Schnorr.Chunked.sign sk
+                    (Random_oracle.Input.Chunked.field full_commitment)
+              }
+          | fee_payer ->
+              fee_payer
+        in
+        let other_parties =
+          Parties.Call_forest.map other_parties ~f:(function
+            | ({ body = { public_key; use_full_commitment; _ }
+               ; authorization = Signature _
+               } as party :
+                Party.t)
+              when Public_key.Compressed.equal public_key pk_compressed ->
+                let commitment =
+                  if use_full_commitment then full_commitment
+                  else transaction_commitment
+                in
+                { party with
+                  authorization =
+                    Signature
+                      (Schnorr.Chunked.sign sk
+                         (Random_oracle.Input.Chunked.field commitment))
+                }
+            | party ->
+                party)
+        in
+        { fee_payer; other_parties; memo }
+      in
+      let parties : Parties.t =
+        sign_all { fee_payer; other_parties = ps; memo }
+      in
+      Ledger.with_ledger ~depth:ledger_depth ~f:(fun ledger ->
+          let account =
+            Account.create account_id
+              Currency.Balance.(
+                Option.value_exn (add_amount zero (Currency.Amount.of_int 500)))
+          in
+          let _, _ =
+            Ledger.get_or_create_account ledger account_id account
+            |> Or_error.ok_exn
+          in
+          let target_ledger = apply_parties ledger [ parties ] in
```

Comment:
use `check_parties_with_merges_exn` instead? It checks that Ledger.apply_transaction matches the what the one in the proof and also does the merge proof of parties